src/webgpu/api/operation/resource_init/check_texture/by_sampling.ts

```diff
@@ -0,0 +1,149 @@
+    let shaderType = 'f32';
+    if (params.format.indexOf('sint') !== -1) {
+      shaderType = 'i32';
+      ReadbackTypedArray = Int32Array;
+    } else if (params.format.indexOf('uint') !== -1) {
+      shaderType = 'u32';
+      ReadbackTypedArray = Uint32Array;
+    }
+
+    const componentOrder = rep.componentOrder;
+    const componentCount = componentOrder.length;
+    const indexExpression =
+      componentCount === 1
+        ? componentOrder[0].toLowerCase()
+        : componentOrder.map(c => c.toLowerCase()).join('') + '[i]';
+
+    const _xd = '_' + params.dimension;
+    const _multisampled = params.sampleCount > 1 ? '_multisampled' : '';
+    const computePipeline = t.device.createComputePipeline({
+      computeStage: {
+        entryPoint: 'main',
+        module: t.device.createShaderModule({
+          code: `
+            [[block]] struct Constants {
+              [[offset(0)]] level : i32;
+            };
+
+            [[set(0), binding(0)]] var<uniform> constants : Constants;
+            [[set(0), binding(1)]] var<uniform_constant> myTexture : texture${_multisampled}${_xd}<${shaderType}>;
+
+            [[block]] struct Result {
+              [[offset(0)]] values : [[stride(4)]] array<${shaderType}>;
+            };
+            [[set(0), binding(3)]] var<storage_buffer> result : Result;
+
+            [[builtin(global_invocation_id)]] var<in> GlobalInvocationID : vec3<u32>;
+
+            [[stage(compute)]]
+            fn main() -> void {
+              var flatIndex : u32 = ${width}u * GlobalInvocationID.y + GlobalInvocationID.x;
+              flatIndex = flatIndex * ${componentCount}u;
+              var texel : vec4<${shaderType}> = textureLoad(
+                myTexture, vec2<i32>(GlobalInvocationID.xy), constants.level);
+
+              for (var i : u32 = flatIndex; i < flatIndex + ${componentCount}u; i = i + 1) {
+                result.values[i] = texel.${indexExpression};
+              }
+              return;
+            }`,
+        }),
+      },
+    });
+
+    for (const slice of slices) {
+      const ubo = t.device.createBuffer({
+        mappedAtCreation: true,
+        size: 4,
+        usage: GPUBufferUsage.UNIFORM | GPUBufferUsage.COPY_DST,
+      });
+      new Int32Array(ubo.getMappedRange(), 0, 1)[0] = level;
+      ubo.unmap();
+
+      const byteLength =
+        width * height * ReadbackTypedArray.BYTES_PER_ELEMENT * rep.componentOrder.length;
+      const resultBuffer = t.device.createBuffer({
+        size: byteLength,
+        usage: GPUBufferUsage.STORAGE | GPUBufferUsage.COPY_SRC,
+      });
+
+      const bindGroup = t.device.createBindGroup({
+        layout: computePipeline.getBindGroupLayout(0),
+        entries: [
+          {
+            binding: 0,
+            resource: { buffer: ubo },
+          },
+          {
+            binding: 1,
+            resource: texture.createView({
+              baseMipLevel: 0,
+              mipLevelCount: params.mipLevelCount,
```

Comment:
don't we want to check each mipLevel independently? Right now I believe only the first one will be checked.

Reply:
the level is passed in as a uniform